app/routes.py

```diff
@@ -1 +1,268 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, jsonify, abort, make_response, request
+from datetime import date
+from app import db
+import requests
+import os
+
+from app.models.task import Task
+from app.models.goal import Goal
+
+# - - - - - - - - - - - - - - - - - #
+###~~~---- helper function ----~~~###
+# - - - - - - - - - - - - - - - - - #
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    model = cls.query.get(model_id)
+
+    if not model:
+        abort(make_response({"message": f"Sorry, {cls.__name__} {model_id} does not exist"}, 404))
+
+    return model
+
+# - - - - - - - - - - - - - - - - - #
+###~~~--- task model routes ---~~~###
+# - - - - - - - - - - - - - - - - - #
+
+task_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+@task_bp.route("", methods=['POST'])
+
+# define a route for creating a task
+def create_task():
+    request_body = request.get_json()
+
+    try:
+        new_task = Task(
+            title = request_body["title"],
+            description = request_body["description"],
+        )
+    except:
+        abort(make_response({"details": f"Invalid data"}, 400))
+
+    db.session.add(new_task)
+    db.session.commit()
```

Comment:
Like we validated models, could we also make a general function that validates request bodies? Here's an example:
```python
def validate_request_body(request_body, keys):
    for key in keys:
        if not request_body.get(key): 
            abort(make_response({
                'Invalid Data': f'missing key: {key}'
            }, 400))

    return True
```
We can pass in the `request_body` and a list of strings that are `keys` and then check to see if those keys are present.